lighthouse-cli/test/smokehouse/smokehouse.js

```diff
@@ -1,154 +1,225 @@
-#!/usr/bin/env node
 /**
  * @license Copyright 2016 Google Inc. All Rights Reserved.
  * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
  * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
  */
 'use strict';
 
+const log = require('lighthouse-logger');
+const cliLighthouseRunner = require('./lighthouse-runners/cli.js').runLighthouse;
+const getAssertionReport = require('./report-assert.js');
+const LocalConsole = require('./local-console.js');
+const ConcurrentMapper = require('./concurrent-mapper.js');
+
 /* eslint-disable no-console */
 
-const fs = require('fs');
-const spawnSync = require('child_process').spawnSync;
-const yargs = require('yargs');
-const log = require('lighthouse-logger');
-const rimraf = require('rimraf');
+/** @typedef {import('./lighthouse-runners/child-process-error.js')} ChildProcessError */
 
-const {collateResults, report} = require('./smokehouse-report.js');
-const assetSaver = require('../../../lighthouse-core/lib/asset-saver.js');
+// The number of concurrent (`!runSerially`) tests to run if `jobs` isn't set.
+const DEFAULT_CONCURRENT_RUNS = 5;
+const DEFAULT_RETRIES = 1;
 
-const PROTOCOL_TIMEOUT_EXIT_CODE = 67;
-const RETRIES = 3;
+/**
+ * @typedef {object} SmokehouseOptions
```

Comment:
does ` {object}` do anything here if we specify the properties right below it?

Reply:
> does ` {object}` do anything here if we specify the properties right below it?

no, and it reads better without it (I assume that's also what you were after?)